src/client.ts

```diff
@@ -9,25 +9,22 @@ import { CommandHandlerConfig } from './types/CommandHandlerConfig.js'
 import { EventHandlerConfig } from './types/EventHandlerConfig.js'
 import { RuntimeConfiguration } from './utils/RuntimeConfiguration.js'
 
-export default class Bot extends Client {
+export default class Bot {
```

Comment:
Can you remove `export default` as it is unnecessary?
If we need it for unit test, it is better to just `export` not default export

Reply:
done. also moved `Bot` to a different file because right now a test tries to initialize the bot immediately upon importing `client.ts`

Reply:
Should we rename `client.ts` to `index.ts`?
Also, please change test file name too. (`client.spec.ts` -> `bot.spec.ts`)

Reply:
@Leomotors can i do it later as it will conflict with #112